packages/block-editor/src/components/block-tools/zoom-out-mode-inserters.js

```diff
@@ -86,6 +79,8 @@ function ZoomOutModeInserters() {
 			hoveredBlockClientId === previousClientId ||
 			hoveredBlockClientId === nextClientId;
 
+		const { showInsertionPoint } = unlock( blockEditorDispatch );
```

Comment:
I think we can unlock the action once instead of doing it in a loop.\

This should also work:

```
const { showInsertionPoint } = unlock( useDispatch( blockEditorStore ) );
```

Reply:
When I did it earlier in the component I got this error from the linter:

`Variables should not be assigned until just prior its first reference. An early return statement may leave this variable unused`

And when it's later I get:

`React Hook "useDispatch" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?`

That's why I split it in two!

Reply:
I think it only triggers that error when using `unlock`. It doesn't do it without it. Probably worth investigating separately.

Now that unlocking isn't needed, you can safely do `const { showInsertionPoint } = useDispatch( blockEditorStore );` earlier in the component.

Reply:
@scruffian, it looks like you already merged PR, so here's the quick follow-up - #63936

I'm also curious if you tested this with page content focus - https://github.com/WordPress/gutenberg/pull/63934#issuecomment-2250226819.

Reply:
Thanks for the follow up. I was keen to merge so that we can move forward with https://github.com/WordPress/gutenberg/pull/63896.

I'll make sure to also look at page content focus on https://github.com/WordPress/gutenberg/pull/63896.